data/src/main/java/me/androidbox/data/alarm_manager/AlarmReceiver.kt

```diff
@@ -1,14 +1,47 @@
 package me.androidbox.data.alarm_manager
 
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import me.androidbox.data.R
 
 class AlarmReceiver : BroadcastReceiver() {
-
+    companion object {
+        const val CHANNEL_ID = "alarm_event_id"
+        const val CHANNEL_NAME = "alarm_event_name"
+        const val EXTRA_MESSAGE = "extra_message"
+    }
     override fun onReceive(context: Context?, intent: Intent?) {
-        intent?.getStringExtra("EXTRA_MESSAGE")?.let { message ->
-            /* TODO Add Notification here */
+        intent?.getStringExtra(EXTRA_MESSAGE)?.let { message ->
+            val notificationManager = context?.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+            if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                val channel = NotificationChannel(
+                    CHANNEL_ID,
+                    CHANNEL_NAME,
+                    NotificationManager.IMPORTANCE_HIGH
+                )
+
+                notificationManager.createNotificationChannel(channel)
+            }
+
+            val notificationIntent = Intent(context, AlarmReceiver::class.java)
+            val pendingIntent = PendingIntent.getActivity(context, 1, notificationIntent, PendingIntent.FLAG_IMMUTABLE)
+            val notification = NotificationCompat.Builder(context, CHANNEL_ID)
+                .setContentTitle(context.getString(R.string.notification_title))
+                .setSmallIcon(R.drawable.bell)
+                .setContentText(message)
+                .setAutoCancel(true)
+                .setContentIntent(pendingIntent)
+                .build()
+
+            notificationManager.notify(1, notification)
```

Comment:
What if a user has it set so it has a reminder, an event and 1 task at the same time?

With .notify(1  you are setting the same id always, so the notification will get overridden
You could use something like:

1st - You should also pass the event/reminder/task id
2nd - Use notify with the id being the hashcode of the item (Ej: reminder.id.hashcode())

This makes it so each notification is unique to an item´s id

Reply:
Also, when a notification get´s clicked, you should send the user to the event/reminder/task detail screen

Reply:
@MKiperszmid I understand that we send reminders to each agenda item (event, tasks, and reminders). And the alarm manager gets this information from room DB.

1) User creates a new event with the start, end, time, date, reminder time, etc, and saves that to the DB
2) This will be uploaded to the API as well.
3) This is the bit I am not sure about, if the alarm manager gets the information from the DB, each time a reminder is updated or created. How does it know it needs to do that? 

Could it be after a insert or update has happened the alarm manager has to query the DB.

Reply:
Simply see AlarmManager as something that will notify your app at a given timestamp and not more. You can then receive and respond to this trigger by using a `BroadcastReceiver`. This will give you an Intent from which you can extract extras, like the agenda item ID the alarm was triggered for. You then take this ID and load the item from the DB to show the notification in the broadcast receiver.
There's no need to listen to insertions, etc. But you will need to update the alarm if the user updates the reminder type or the agenda item time